vector-im/element-android
# FTUE - Email input and verification

## Type of change


## Content

Introduces new dedicated _email entry_ and _waiting for verification_ fragments to the FTUE onboarding flow.

- Lifts the 401 verification polling error handling logic out of the fragment and into the `RegisterActionHandler`, adds tests around the polling. 
- A loading spinner is shown only when returning to the verification fragment rather than on the initial navigation
- A dedicated resend email button has been added 

## Motivation and context

Fixes #5278 

## Screenshots / GIFs

#### Flow 

| | 
| --- |
|![after-signup-e2e](https://user-images.githubusercontent.com/1848238/165780534-6bc9905d-368d-4440-bdfa-75c21db2b92e.gif)


#### Phones

| INPUT | INPUT WITH CONTENT | WAITING | WAITING RETURN TO SCREEN |
| --- | --- | --- | --- |
|![Screenshot_20220428_151111](https://user-images.githubusercontent.com/1848238/165773033-1160f8d1-0c8c-49eb-a721-bca8fe37d01c.png)|![Screenshot_20220428_151127](https://user-images.githubusercontent.com/1848238/165773031-7141aece-28b5-4502-a56a-d7ee87be8fd4.png)|![Screenshot_20220428_151135](https://user-images.githubusercontent.com/1848238/165773029-63fc4b53-4bf9-4bea-8f93-3f2d7abb8725.png)|![Screenshot_20220428_151145](https://user-images.githubusercontent.com/1848238/165773026-175b996b-3fe8-43bc-9c43-6ccbc2ff1389.png)

#### Small Phone

| INPUT | WAITING |
| --- | --- |
|![Screenshot_20220428_151300](https://user-images.githubusercontent.com/1848238/165773464-bc8165d7-79cd-4e0f-b135-2ba5696785cf.png)|![Screenshot_20220428_151312](https://user-images.githubusercontent.com/1848238/165773462-cb0d7de7-7570-4842-9f98-f652eaa581fa.png)


#### Tablet

| INPUT | WAITING | 
| --- | --- |
|![Screenshot_20220428_151415](https://user-images.githubusercontent.com/1848238/165772621-19159eea-8d66-405b-af83-c855b699d43b.png)|![Screenshot_20220428_151426](https://user-images.githubusercontent.com/1848238/165772620-56d3af2b-1727-485d-9ef0-5f2e963db48d.png)
![Screenshot_20220428_151409](https://user-images.githubusercontent.com/1848238/165772623-79d1c5d6-873e-48f5-ad16-0c86e91e9092.png)|![Screenshot_20220428_151433](https://user-images.githubusercontent.com/1848238/165772611-66db8637-6f21-49ef-b318-9f6da9591837.png)


## Tests

- Enable the combined register feature flag
- Create an account of matrix.org
- Notice the new email verification screens

## Tested devices

- OS version(s): 28, 29, 31
